File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/objectivecm/template-const.m

#import <Foundation/Foundation.h>
#import "__CLASS_NAME__.h"

// == START OF EDITABLE AREA FOR ABOVE ==
// == END OF EDITABLE AREA FOR ABOVE ==

// THIS CLASS WAS AUTOMATICALLY CREATED BY MTOOL. DON'T EDIT EXCEPT THE EDITABLE AREA.
// [JP] このクラスは自動生成されています。編集可能領域以外は変更しないで下さい。

NSString *const __CLASS_NAME__TextArray[] = {
__CONST_STRING_DEFINITIONS__
};
int const __CLASS_NAME__TextArraySize = __DEFINE_COUNT__;

@implementation __CLASS_NAME__Conversion
{
}

+ (NSString*) Get__CLASS_NAME__Text:(NSInteger) val
{
    if (val >= 0 && val < __CLASS_NAME__TextArraySize) {
        return __CLASS_NAME__TextArray[val];
    }
    return nil;
}
+ (__CLASS_NAME__) Get__CLASS_NAME__:(NSString*) enum_value
{
    for(int index = 0 ; index < __CLASS_NAME__TextArraySize ;index++) {
        if ([__CLASS_NAME__TextArray[index] compare:enum_value]) {
            return index;
        }
    }
    return -1;
}

// == START OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==
// == END OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==

@end

// == START OF EDITABLE AREA FOR BOTTOM ==
// == END OF EDITABLE AREA FOR BOTTOM ==

File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/swift/template-const.swift

// == START OF EDITABLE AREA FOR ABOVE ==
// == END OF EDITABLE AREA FOR ABOVE ==

// THIS CLASS WAS AUTOMATICALLY CREATED BY MTOOL. DON'T EDIT EXCEPT THE EDITABLE AREA.
// [JP] このクラスは自動生成されています。編集可能領域以外は変更しないで下さい。

public enum __CLASS_NAME__:Int {
    case
__CONST_DEFINITIONS__
}

public let __CLASS_NAME__TextArray = [
__CONST_STRING_DEFINITIONS__
];

public class __CLASS_NAME__Conversion
{
    public static func Get__CLASS_NAME__Text(val:__CLASS_NAME__) -> String
    {
        if (val.rawValue >= 0 && val.rawValue < __CLASS_NAME__TextArray.count) {
            return __CLASS_NAME__TextArray[val.rawValue];
        }
        return "";
    }
    public static func Get__CLASS_NAME__(enum_value:Any?) -> __CLASS_NAME__
    {
        let enum_string:String = ProxyClientDataConversionUtil.ConvToString(value:enum_value);
        for index in 0 ..< __CLASS_NAME__TextArray.count {
            if (enum_string == __CLASS_NAME__TextArray[index]) {
                return __CLASS_NAME__(rawValue:index)!;
            }
        }
        return __CLASS_NAME__.UNKNOWN;
    }

// == START OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==
// == END OF EDITABLE AREA FOR ADDITIONAL CLASS IMPLEMENTATION DEFINITION ==

}

// == START OF EDITABLE AREA FOR BOTTOM ==
// == END OF EDITABLE AREA FOR BOTTOM ==
